homeassistant/components/smartthings/number.py

```diff
@@ -0,0 +1,77 @@
+"""Support for number entities through the SmartThings cloud API."""
+
+from __future__ import annotations
+
+from pysmartthings import Attribute, Capability, Command, SmartThings
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import FullDevice, SmartThingsConfigEntry
+from .const import MAIN
+from .entity import SmartThingsEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SmartThingsConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Add number entities for a config entry."""
+    entry_data = entry.runtime_data
+    async_add_entities(
+        SmartThingsNumberEntity(entry_data.client, device)
+        for device in entry_data.devices.values()
+        if Capability.CUSTOM_WASHER_RINSE_CYCLES in device.status[MAIN]
+    )
+
+
+class SmartThingsNumberEntity(SmartThingsEntity, NumberEntity):
+    """Define a SmartThings number."""
+
+    _attr_translation_key = "washer_rinse_cycles"
```

Comment:
Anticipating there are other capabilities that will use the Number platform in the future, would it make sense to move the things specific to `washer_rince_cycles` to parameters instead of hardcoding? Or name the class specific to what it is? (`WasherRinseCycleNumberEntity`).

Reply:
Updated it to set it to the specific class. I do want to make them more generic but I didn't have more numbers just yet